### ISSUE TITLE
Entries field modal incorrectly populated in multi-site setup

### ISSUE THREAD

**Issue by baumberger** [craftcms/cms#7015]
### Description

The modal to select entries in an entries field is populated with incorrect entries when changing the source section of the field in a multi-site setup.

### Steps to reproduce

1. Create multi-site setup with _Site A_ and _Site B_
2. Create structure section _Section A_ that is only available on _Site A_ (create some entries in section)
3. Create structure section _Section B_ that is only available on _Site B_ (create some entries in section)
4. Create entries field e.g. _Related Entries_ and select _Section A_ in _Sources_
5. Attach _Related Entries_ field to users
6. Edit a user -> _Related Entries_ correctly shows entries from _Section A_
7. Edit _Related Entries_ field and change _Sources_ to _Section B_
8. Edit a user -> _Related Entries_ incorrectly show entries from _Section A_

### Observations
- The bug can disappear if after step (7) you go to _admin/entries_ and click through all sites and structures
- Sometimes the bug then also reappears
- Occasionally (so far did not find out how to reproduce), opening the select entries modal leads to an _Internal Server Error_ (error logged to `web.log` is pasted below)
- All testing was done with a single admin account

### Note
If you need additional information, I will gladly provide you with what you need.

### Additional info

- Craft version: 3.5.12.1
- PHP version: 7.4.10
- Database driver & version: MySQL 5.5.5
- Plugins & versions: None

```
2020-10-15 19:08:12 [-][1][-][error][yii\web\HttpException:403] yii\web\ForbiddenHttpException: User not permitted to access this source in /home/ubuntu/sites/bug-report/vendor/craftcms/cms/src/controllers/ElementIndexesController.php:399
Stack trace:
#0 /home/ubuntu/sites/bug-report/vendor/craftcms/cms/src/controllers/ElementIndexesController.php(100): craft\controllers\ElementIndexesController->source()
#1 /home/ubuntu/sites/bug-report/vendor/yiisoft/yii2/base/Controller.php(178): craft\controllers\ElementIndexesController->beforeAction()
#2 /home/ubuntu/sites/bug-report/vendor/craftcms/cms/src/web/Controller.php(189): yii\base\Controller->runAction()
#3 /home/ubuntu/sites/bug-report/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction()
#4 /home/ubuntu/sites/bug-report/vendor/craftcms/cms/src/web/Application.php(274): yii\base\Module->runAction()
#5 /home/ubuntu/sites/bug-report/vendor/craftcms/cms/src/web/Application.php(577): craft\web\Application->runAction()
#6 /home/ubuntu/sites/bug-report/vendor/craftcms/cms/src/web/Application.php(253): craft\web\Application->_processActionRequest()
#7 /home/ubuntu/sites/bug-report/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest()
#8 /home/ubuntu/sites/bug-report/web/index.php(22): yii\base\Application->run()
#9 {main}
2020-10-15 19:08:11 [-][1][-][info][application] $_GET = [
    'p' => 'admin/actions/element-indexes/get-elements'
    'v' => '1602781691508'
]
```

**Comment by baumberger** [comment]
This seems fixed with #6968. Very happy.

**Comment by brandonkelly** [comment]
Glad that fixed it for you!
